### ISSUE TITLE
UserWarning: implicit cast from 'char *' to a different pointer type

### ISSUE THREAD

**Issue by mgedmin** [pyca/cryptography#3156]
I upgraded a bunch of packages recently, including `cryptography` (to 1.5) and `cffi` (to 1.8.3), and now every time I run `ansible` I see this warning:

```
/home/mg/.venv/local/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/backend.py:307: UserWarning: implicit cast from 'char *' to a different pointer type: will be forbidden in the future (check that the types are as you expect; use an explicit ffi.cast() if they are correct)
  buf
```

(I'm just guessing that cffi is involved in this.)

That line is part of this statement:
```python
        res = self._lib.PKCS5_PBKDF2_HMAC(
            key_material,
            len(key_material),
            salt,
            len(salt),
            iterations,
            evp_md,
            length,
            buf
        )
```

**Comment by reaperhulk** [comment]
#3155 was submitted yesterday to resolve this. It will be in a release later this week.

**Comment by reaperhulk** [comment]
(Resolved in 1.5.1)
